core/arch/arm/crypto/ghash-ce-core_a64.S

```diff
@@ -11,11 +11,10 @@
 #define CPU_LE(x...)	x
 
 /*
- * If the lower half of CTR is initialized with zeroes or a low value we
- * can expect that the upper half will remain unchanged. As an optimization
- * make the code to increase the upper half optional.
+ * Only increase the lowest quarter, that is, 32-bits of the counter. If
+ * it wraps it must not propagate into the upper bits.
  */
-#define INC_HALF_CTR	0
+#define INC_QUART_CTR	1
```

Comment:
IIUC the meaning of the setting was "increment the upper half of the counter", and you're changing it to "increment the lower quarter only". In both cases it is a bit complicated... why keep a configuration flag at all?

Reply:
No, it was "increment the lower half of the counter". Anyway, it's all a bit complicated. I kept it because it makes it easier to see where the counter is manipulated. That can be achieved with a bunch of comments too, but this minimizes the delta from the imported code.

Reply:
> No, it was "increment the lower half of the counter".

Hmmm... so prior to this change we did NOT increment the lower half? (it was 0). Sounds weird, the lower half are the less significant bits aren't they? The ones that matter?